pkg/deploy/strategy/util/util.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// RecordConfigEvent records an event for the deployment config referenced by the
+// deployment.
+func RecordConfigEvent(client kclient.EventNamespacer, deployment *kapi.ReplicationController, decoder runtime.Decoder, eventType, reason, msg string) {
+	t := unversioned.Time{Time: time.Now()}
+	var obj runtime.Object = deployment
+	if config, err := deployutil.DecodeDeploymentConfig(deployment, decoder); err != nil {
+		glog.Errorf("Unable to decode deployment config from %s/%s: %v", deployment.Namespace, deployment.Name, err)
+		return
```

Comment:
At this point we don't care about the decoding error. Logging is fine but we should not prevent the creation of the event.